### ISSUE TITLE
Not able to replicate paper results

### ISSUE THREAD

**Issue by andresespinosapc** [i-machine-think/seq2attn#1]
Hi! I've been trying to replicate the results obtained in [Transcoding compositionally: using attention to find more generalizable solutions](https://arxiv.org/abs/1906.01234), but I haven't been successful. In the lookup tables dataset, I ran 10 experiments with different random seeds. Only 4 obtained more than 97% sequence accuracy (2 obtained 100%) on the heldout_tables task. The command line options I used are:

```python
{
    'full_attention_focus': True,
    'sample_train': 'gumbel_st',
    'attn_vals': 'embeddings',
    'batch_size': 1,
    'embedding_size': 256,
    'hidden_size': 256,
    'n_layers': 1,
    'rnn_cell': 'gru',
    'attention': 'pre-rnn',
    'attention_method': 'mlp',
    'max_len': 50,
    'dropout_p_encoder': 0.5,
    'dropout_p_decoder': 0.5,
}
```

The rest of the options are left as default.

**Comment by KrisKorrel** [comment]
Hi Andres! 
I have updated the example script a bit. A quick experiment shows this script to achieve 100% on holdout tables pretty consistently.
Have you checked:
- The number of epochs for which you train. I do not know what number would be optimal, tbh.
- Whether you save the right model (with the lowest validation loss) and loaded the correct version when you evaluate?

**Comment by andresespinosapc** [comment]
Yes, I trained for 100 epochs and plotted the sequence accuracy for every epoch.

Here's the mean sequence accuracy for each epoch (of 5 runs with different random seeds): 

![Screenshot_20200407_184226](https://user-images.githubusercontent.com/8888934/78726267-8913ef00-78ff-11ea-8b6b-8d8f646a1411.png)

(2 out of 5 reached 100%)

**Comment by andresespinosapc** [comment]
I was using some incorrect hyperparameters. Now, 4 out of 5 experiments reach 100%. Thanks
